### PR TITLE
Improve deprecation log messages

### PR DESCRIPTION
**Pull request by lawrence-forooghian**
Add more detail that addresses what the user might have been trying to achieve with their use of the deprecated API, and more detail about how to change their code.

### CODE REVIEW

File: src/common/lib/util/defaults.ts [hunk 1]
```diff
@@ -219,12 +219,15 @@ export function normaliseOptions(options: DeprecatedClientOptions): NormalisedCl
 
     /* emit an appropriate deprecation warning */
     if (options.environment) {
-      Logger.deprecatedWithMsg(
-        'fallbackHostsUseDefault',
-        'There is no longer a need to set this when the environment option is also set since the library will now generate the correct fallback hosts using the environment option.'
+      Logger.deprecated(
+        'The `fallbackHostsUseDefault` client option',
+        'Are you using this client option to force the library to make use of fallback hosts even though you’ve set the `environment` client option? If so, this is no longer necessary: remove your usage of the `fallbackHostsUseDefault` client option and the library will then automatically choose the correct fallback hosts to use for the specified environment.'
```
Comment (lmars) [inline src/common/lib/util/defaults.ts:224]: I'm not a fan of this being a question, how about:

```suggestion
        'If you are you using this client option to force the library to use default fallback hosts even though you’ve set the `environment` client option, then this is no longer necessary: remove your usage of the `fallbackHostsUseDefault` client option and the library will then automatically choose the correct fallback hosts to use for the specified environment.'
```

Comment (lawrence-forooghian) [reply src/common/lib/util/defaults.ts]: Thanks, have updated.

**Review by VeskeR — APPROVED** [review]
LGTM

File: src/common/lib/util/defaults.ts (hunk 1 above)
Comment (lawrence-forooghian) [reply src/common/lib/util/defaults.ts]: Going to resolve this conversation.

**Pull request merged**